maps/us-healthcare-system-capacity/js/main.js

```diff
@@ -293,33 +306,37 @@ function getProperty(theIndicator) {
   return indicatorProperty;
 }
 
-function getMinMax() {
-  return minMax[type][getProperty(indicator)];
+function getBreaks() {
+  return breaks[type][getProperty(indicator)];
 }
 
 function setFillPaintStyle(layerName) {
-  var colors = indicators[indicator].colors;
-  var minMaxValues = getMinMax();
-  setLegend(colors, minMaxValues);
-
-  map.setPaintProperty(layerName, "fill-color", [
-    "interpolate",
-    ["linear"],
-    ["number", ["get", getProperty(indicator)], minMaxValues[0]],
-    minMaxValues[0],
-    colors[0],
-    minMaxValues[1],
-    colors[1]
-  ]);
+  var colorsMinMax = indicators[indicator].colors,
+      breaksValues = getBreaks(),
+      palette = interpolate(colorsMinMax),
+      colors = _.map([...Array(breaksValues.length).keys()], function(i) {
+        return palette(i / breaksValues.length);
+      }),
+      style = [
+        "interpolate",
+        ["linear"],
+        ["number", ["get", getProperty(indicator)], breaksValues[0]]
+      ].concat(
+        _.flatten(_.zip(breaksValues, colors))
+      );
```

Comment:
This code:
- constructs a "palette" which you an call with a number between 0 and 1 to do interpolation between the start and end color
- creates colors based on counting 0..N, where N is the number of breaks, and assigning a color to each of those.
- style then uses a zipped up version to append the mapbox style stops. The `_.zip` method creates a list of breaks to colors, in the form `[[break1, color1],[break2, color2], ...]`. We need those values in a straight list so that's what `flatten` does - converts the above into `[break1, color1, break2, color2, ... ]`